### ISSUE TITLE
Performance-focused defaults

### ISSUE THREAD

**Issue by SomeoneSerge** [numtide/nixpkgs-unfree#5]
**Is your feature request related to a problem? Please describe.**

I only use unfree when I build for sci-comp applications. This virtually always means that I also want:

```nix
cudaSupport = true

blas = prev.blas.override {
    blasProvider = final.mkl;
};

lapack = prev.lapack.override {
    lapackProvider = final.mkl;
};
```

There could be other options that people commonly use together with `allowUnfree`.
At least the ones listed cause a significant amount of rebuilds.

**Describe the solution you'd like**

Choose an acceptable set of "defaults", including the `config` and an overlay, that are common for sci-comp purposes.
Build and cache nixpkgs with these defaults. 

**Describe alternatives you've considered**

There seem to be two purposes to this flake:
- Being the single root instance for all nixpkgs that have `unfree` packages enabled
- Providing the infrastructure for testing and caching the heavy unfree packages

Overriding globally the `cudaSupport` and `blas` might potentially be detrimental to the former. We might want to still expose the "simplest" nixpkgs by default (i.e. `allowUnfree = true` only), but I think it's important that a modest overlay like above does not result in too many cache-misses

**Comment by Mic92** [comment]
Yeah. Having cudaSupport enabled everywhere makes a lot of sense.

**Comment by Mic92** [comment]
Related https://github.com/markuskowa/NixOS-QChem/blob/master/overlays/cuda.nix

**Comment by zimbatm** [comment]
Sounds good. Ideally, nixpkgs gets a new cudaSupport global config flag so we don't have to maintain an overlay.

**Comment by zimbatm** [comment]
I also found https://github.com/nix-community/nix-data-science which might have some interesting learnings in there.

**Comment by zimbatm** [comment]
Related issue that needs to be addressed to make this credible: https://discourse.nixos.org/t/using-cuda-enabled-packages-on-non-nixos-systems/17788
